autoreporter_addons/ssl_checks/reporter.py

```diff
@@ -0,0 +1,203 @@
+
+logger = utils.build_logger(__name__)
+
+
+with open(os.path.join(os.path.dirname(__file__), "filtered_website_fragments.txt"), "r") as f:
+    FILTERED_WEBSITE_FRAGMENTS = [line.strip() for line in f.readlines() if line]
+
+with open(os.path.join(os.path.dirname(__file__), "filtered_website_fragments_for_bad_redirect.txt"), "r") as f:
+    # These fragments, if occur, mean that we shouldn't treat this website as containing a bad redirect.
+    # For instance, if Cloudflare returned HTTP 200 with a message "Please wait while your request is being verified...",
+    # that doesn't meant that the original website doesn't redirect to https:// - that means only, that our request
+    # got intercepted via Cloudflare.
+    FILTERED_WEBSITE_FRAGMENTS_FOR_BAD_REDIRECT = [line.strip() for line in f.readlines() if line]
+
+
+class SSLChecksReporter(Reporter):  # type: ignore
+    CERTIFICATE_AUTHORITY_INVALID = ReportType("certificate_authority_invalid")
+    NO_HTTPS_REDIRECT = ReportType("no_https_redirect")
+    BAD_CERTIFICATE_NAMES = ReportType("bad_certificate_names")
+    ALMOST_EXPIRED_SSL_CERTIFICATE = ReportType("almost_expired_ssl_certificate")
+    EXPIRED_SSL_CERTIFICATE = ReportType("expired_ssl_certificate")
+
+    @staticmethod
+    def create_reports(task_result: Dict[str, Any], language: Language) -> List[Report]:
+        if task_result["headers"]["receiver"] != "ssl_checks":
+            return []
+
+        domain = task_result["payload"]["domain"]
+        domain_parts = [part for part in domain.split(".") if part]
+        # We do the filtering both in the scanning module and in the reporter so that after changing this setting
+        # the user wouldn't need to wait for the next scan for the configuration to take effect.
+        if domain_parts[0] in ExtraModulesConfig.SUBDOMAINS_TO_SKIP_SSL_CHECKS:
+            return []
+
+        if not isinstance(task_result["result"], dict):
+            return []
+
+        try:
+            response = cached_get(f"https://{domain}")
+            parent_response = cached_get(f"https://{'.'.join(domain_parts[1:])}")
+            if SequenceMatcher(None, response.content, parent_response.content).quick_ratio() >= 0.8:
+                # Do not report misconfigurations if a domain has identical content to a parent domain - e.g.
+                # if we have mail.domain.com with identical content to domain.com, we assume that it's domain.com
+                # which is actually used, and therefore don't report subdomains.
+                return []
+        except Exception:
+            logger.warning(f"Unable to check whether domain {domain} has identical content to parent domain")
+
+        if "response_status_code" in task_result["result"] and "response_content_prefix" in task_result["result"]:
+            response_status_code = task_result["result"]["response_status_code"]
+            response_content_prefix = task_result["result"]["response_content_prefix"]
+
+            if (
```

Comment:
I would split this if into two, in regard to variable in question. It would make if statement little cleaner